File: warmup/salesByMatch.js
function sockMerchant(n, ar) {
  const socks = ar.reduce((result, sock) => {
    if (result[sock]) {
      result[sock]++;
    } else {
      result[sock] = 1;
    }

    return result;
  }, {});

  const pairs = Object.values(socks).reduce(
    (pairs, color) => pairs + Math.floor(color / 2),
    0
  );
  return pairs;
}

sockMerchant(7, [1, 2, 1, 2, 1, 3, 2]);

File: warmup/sales_by_match.py
def matchSocks(n, ar):

    # sort arr into socks by color
    socks = {n: ar.count(n) for n in ar}
    # get number of pairs for each color, sum
    return sum([s//2 for s in socks.values()])


print(matchSocks(7, [1, 2, 1, 2, 1, 3, 2]))
